janitor/functions.py

```diff
@@ -4245,3 +4245,133 @@ def fill_direction(
+        import janitor as jn
+
+        df = pd.DataFrame({'age' : [20, 22, 24, 23, 21, 22],
+                           'ID' : [1,2,3,4,5,6],
+                           'result' : ["pass", "fail", "pass",
+                                       "pass", "fail", "pass"]})
+
+        # Ascending top 3:
+        df.groupby_topk('result', 'age', 3)
+        #       age  ID  result
+        #result
+        #fail   21   5   fail
+        #       22   2   fail
+        #pass   20   1   pass
+        #       22   6   pass
+        #       23   4   pass
+
+        #Descending top 2:
+        df.groupby_topk('result', 'age', 2, {'ascending':False})
+        #       age  ID result
+        #result
+        #fail   22   2   fail
+        #       21   5   fail
+        #pass   24   3   pass
+        #       23   4   pass
+
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = pd.DataFrame(...)
+        df = jn.groupby_topk(
+            df = df,
+            groupby_column_name = 'groupby_column',
+            sort_column_name = 'sort_column',
+            k = 5
+            )
+
+    Method-chaining usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = (
+            pd.DataFrame(...)
+            .groupby_topk(
+            df = df,
+            groupby_column_name = 'groupby_column',
+            sort_column_name = 'sort_column',
+            k = 5
+            )
+        )
+
+    :param df: A pandas dataframe.
+    :param groupby_column_name: Column name to group input dataframe `df` by.
+    :param sort_column_name: Name of the column to sort along the
+        input dataframe `df`.
+    :param k: Number of top rows to return from each group after sorting.
+    :param sort_values_kwargs: Arguments to be passed to sort_values function.
+    :returns: A pandas dataframe with top `k` rows that are grouped by
+        `groupby_column_name` column with each group sorted along the
+        column `sort_column_name`.
+    :raises: ValueError if `k` is less than 1.
+    :raises: ValueError if `groupb y_column_name` does not exist in dataframe.
+    :raises: ValueError if `sort_column_name` does not exist in dataframe.
+    :raises: ValueError if `groupby_column_name` is same as `sort_column_name`.
+    :raises: KeyError if `inplace:True` is present in `sort_values_kwargs`.
+    """
+
+    # Convert the default sort_values_kwargs from None to empty Dict
+    sort_values_kwargs = sort_values_kwargs or {}
+
+    # Check if k is greater than 0.
+    if k < 1:
+        raise ValueError(
+            "Numbers of rows per group to be returned must be greater than 0."
+        )
+
+    # Check if groupby_column_name exists in dataframe.
+    if groupby_column_name not in df.columns:
+        raise ValueError(
+            f"Column {groupby_column_name} does not exist in the dataframe."
+        )
```

Comment:
```suggestion
    if groupby_column_name not in df.columns:
        raise KeyError(
            f"Column {groupby_column_name} does not exist in the dataframe."
        )
```

Actually, I think we have a function that does this already.

It's in [`janitor.utils`](https://github.com/ericmjl/pyjanitor/blob/dev/janitor/utils.py#L317). And curiously, it raises a ValueError (really should be a KeyError).

In any case, can you use the `check_column` in place of the current checks that you have? This way we have fewer lines of code to maintain.

Reply:
I've done as As @samukweku suggested in his review #714 (comment), pandas gives a KeyError by itself when column is not found, so I removed these checks and added tests for checking for KeyError when wrong column is provided.

Should I change it to use check_column and add the two KeyErrors or keep it the way it is now and let pandas give the KeyError? Asking since the reviews regarding these checks are different.

Reply:
Now that I've had a bit of time away from the problem for a bit, I think we should change the ValueError in `check` to a KeyError so that it matches what `pandas` gives us. Consistent error message types help simplify error checking, though our `check` function explicitly states the problem. For now, I'd suggest making everything consistent with using `check`; in a separate PR, we can make the codebase consistently return `KeyError`s by changing the underlying implementation of `check`.

Reply:
I believe I got what you mean, but apologies if I misunderstood (My English is not my strongest point).

I think what you want me to do is to remove the checks in function but keep the checks in tests as `KeyError` so they will be consistent with `check_column` later on? If so, I've done this.

Reply:
No worries, @mphirke - looking back on what I wrote, I realize I wasn't communicating clearly either.

I think what was going through my head back then was this: `janitor.utils.check_column` will check whether column names are present in the dataframe or not for us, and returns a more useful error message than pandas' error message. As such, my proposal is:

1. Replace the lines in which you were checking for the presence of a column with the `check_column` function. 
2. Ignore the fact that `check_column` raises a ValueError. 
3. In the tests you have, for now, raise a ValueError rather than KeyError, because that's what `check_column` raises.
4. We can standardize the error type across the codebase back to KeyError in a separate PR.

Reply:
So it seems I understood it all wrong, my bad. Thanks a lot for clarifying. I've done as you suggested.